packages/ui-components/src/icon/jlicon.tsx

```diff
@@ -0,0 +1,414 @@
+    if (cls) {
+      container.classList.remove(cls);
+    }
+
+    return container;
+  }
+
+  render(host: HTMLElement, props: JLIcon.IProps = {}): void {
+    // TODO: move this title fix to the Lumino side
+    host.removeAttribute('title');
+
+    return ReactDOM.render(<this.react container={host} {...props} />, host);
+  }
+
+  resolveSvg(title?: string): HTMLElement | null {
+    const svgDoc = new DOMParser().parseFromString(
+      this._svgstr,
+      'image/svg+xml'
+    );
+    const svgElement = svgDoc.documentElement;
+
+    // structure of error element varies by browser, search at top level
+    if (svgDoc.getElementsByTagName('parsererror').length > 0) {
+      const errmsg = `SVG HTML was malformed for JLIcon instance.\nname: ${name}, svgstr: ${this._svgstr}`;
+      // parse failed, svgElement will be an error box
+      if (JLIcon._debug) {
+        // fail noisily, render the error box
+        console.error(errmsg);
+        return svgElement;
+      } else {
+        // bad svg is always a real error, fail silently but warn
+        console.warn(errmsg);
+        return null;
+      }
+    } else {
+      // parse succeeded
+      svgElement.dataset.iconid = this._uuid;
+
+      if (title) {
+        Private.setTitleSvg(svgElement, title);
+      }
+
+      return svgElement;
+    }
+  }
+
+  get svgstr() {
+    return this._svgstr;
+  }
+
+  set svgstr(svgstr: string) {
+    this._svgstr = svgstr;
+
+    // associate a unique id with this particular svgstr
+    this._uuid = UUID.uuid4();
+  }
+
+  unrender(host: HTMLElement): void {
+    ReactDOM.unmountComponentAtNode(host);
+  }
+
+  protected _initContainer({
+    container,
+    className,
+    propsStyle,
+    title
+  }: {
+    container: HTMLElement;
+    className?: string;
+    propsStyle?: IIconStyle;
+    title?: string;
+  }) {
+    const classStyle = iconStyle(propsStyle);
+
+    if (className != null) {
+      // override the container class with explicitly passed-in class + style class
+      container.className = classes(className, classStyle);
+    } else if (classStyle) {
+      // add the style class to the container class
+      container.classList.add(classStyle);
+    }
+
+    if (title != null) {
+      container.title = title;
+    }
+  }
+
+  protected _initReact() {
+    const component = React.forwardRef(
+      (
+        {
+          className,
+          container,
+          label,
+          title,
+          tag = 'div',
+          ...propsStyle
+        }: JLIcon.IProps = {},
+        ref: React.RefObject<SVGElement>
+      ) => {
+        const Tag = tag;
+
+        // ensure that svg html is valid
+        const svgElement = this.resolveSvg();
+        if (!svgElement) {
+          // bail if failing silently
+          return <></>;
+        }
+
+        const svgComponent = (
+          <svg
+            {...getReactAttrs(svgElement)}
+            dangerouslySetInnerHTML={{ __html: svgElement.innerHTML }}
+            ref={ref}
```

Comment:
Adding `data-icon={this.name}` fixes the notebook tests for me locally.  I think we should keep this data attribute.

Reply:
I've gone back and forth on this. Currently, `data-icon` has been dumped in favor of `data-iconid` (see the `JLIcon.resolveSvg` method), but since `data-iconid` is a uuid, you can't use it for querying/pulling out the icon node the same way you could use `data-icon`. I suppose for now there can just be both `data-icon` and `data-iconid` and we can sort out which should remain later.